pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/PendingAppendingTxnBufferTask.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.transaction.buffer.impl;
+
+import io.netty.buffer.ByteBuf;
+import java.util.concurrent.CompletableFuture;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.bookkeeper.mledger.Position;
+import org.apache.pulsar.client.api.transaction.TxnID;
+
+@Getter
+@AllArgsConstructor
+public class PendingAppendingTxnBufferTask {
+
+    private final TxnID txnId;
+    private final long sequenceId;
+    private final ByteBuf buffer;
+    private CompletableFuture<Position> pendingPublishFuture;
+}
```

Comment:
It seems that this class is only used in TopicTransactionBuffer and could be a Java record defined inside TopicTransactionBuffer.

Reply:
Changed it to a record